crates/sql-sink/src/main.rs

```diff
@@ -27,49 +29,75 @@ const BACKOFF_MAX: Duration = Duration::from_secs(3600 * 24);
 
 #[connector(sink)]
 async fn start(config: SqlConfig, mut stream: impl ConsumerStream) -> Result<()> {
-    let mut sink = start_sink(config).await?;
-    while let Some(item) = stream.next().await {
-        let operation: Operation = serde_json::from_slice(item?.as_ref())?;
-        trace!(?operation);
-        sink.send(operation).await?;
+    let mut backoff = backoff_init()?;
+    let mut sink = start_sink(&mut backoff, &config).await?;
+
+    info!("Starting to process records");
+
+    while let Some(item_result) = stream.next().await {
+        match item_result {
+            Ok(item) => match serde_json::from_slice::<Operation>(item.as_ref()) {
+                Ok(operation) => {
+                    trace!(?operation);
+                    while let Err(e) = sink.send(operation.clone()).await {
+                        error!("Error sending operation to sink: {}", e);
+                        sink = start_sink(&mut backoff, &config).await?;
+                    }
+                }
+                Err(err) => {
+                    error!("Failed to deserialize operation: {}", err);
+                    backoff_and_wait(&mut backoff).await?;
+                    continue;
+                }
+            },
+            Err(err) => {
+                error!("Error reading from stream: {}", err);
+                backoff_and_wait(&mut backoff).await?;
+            }
+        }
     }
 
+    info!("Connection dropped, shutting down");
+
     Ok(())
 }
 
-async fn start_sink(config: SqlConfig) -> Result<LocalBoxSink<Operation>> {
-    let mut backoff = backoff_init()?;
+async fn backoff_and_wait(backoff: &mut ExponentialBackoff) -> Result<()> {
+    let wait = backoff.wait();
+    if wait < BACKOFF_MAX {
+        warn!("Retrying in {}", humantime::format_duration(wait));
+        async_std::task::sleep(wait).await;
+        Ok(())
```

Comment:
Something like "tried 10 times,bailing"

Reply:
It's bailing after 42 times:

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=55fd18ddcd2fb04d2b3dd1a4866ec822

Do you think is enough?

Reply:
That seems excessive since its exponential.  It probably means something is hosed badly or wrong config. Alternatively this could be user config

Reply:
Do you think that it's better to use lower BACKOFF_MAX to have fewer attempts, maybe making possible to create e2e for the failing case.

Reply:
I added backoff_max and backoff_min in sql config, the default now is 60, but user can change it now.

btw, should we use kerbal-case or snake_case? I am using snake_case because I checked that http-sink is using like that, but seems that kerbel-case is better for yaml